base/libgit2/oid.jl

```diff
@@ -90,8 +90,8 @@ function Base.cmp(id1::GitShortHash, id2::GitShortHash)
                   Ref(id1.hash), Ref(id2.hash), min(id1.len, id2.len)))
     return c == 0 ? cmp(id1.len, id2.len) : c
 end
-Base.cmp(id1::GitHash, id2::GitShortHash) = cmp(GitShortHash(id1, OID_RAWSZ), id2)
-Base.cmp(id1::GitShortHash, id2::GitHash) = cmp(id1, GitShortHash(id2, OID_RAWSZ))
+Base.cmp(id1::GitHash, id2::GitShortHash) = cmp(GitShortHash(id1, OID_HEXSZ), id2)
+Base.cmp(id1::GitShortHash, id2::GitHash) = cmp(id1, GitShortHash(id2, OID_HEXSZ))
```

Comment:
guess a comparison that depended on the latter half of the full input would have given differing results here, but we didn't have any such cases?